### PR TITLE
[Test] Test state_dict and loss modules

### PR DESCRIPTION
**Pull request by vmoens**
depends on https://github.com/pytorch-labs/tensordict/pull/528

### CODE REVIEW

**Review by matteobettini — APPROVED** [review]
LGTM
